custom_components/econet300/number.py

```diff
@@ -126,15 +126,25 @@ async def async_set_native_value(self, value: float) -> None:
         self.async_write_ha_state()
 
 
-def can_add(key: str, coordinator: EconetDataCoordinator):
+def can_add_number_entity(key: str, coordinator: EconetDataCoordinator):
     """Check if a given entity can be added based on the availability of data in the coordinator."""
     try:
+        # Check for unsupported controllers
+        controller_id = coordinator.data.get("sysParams", {}).get("controllerID", "")
```

Comment:
I would suggest u to add controller param to coordinator  instead of geting data thrtough other data, that increses % of posible error also less clear in structure

Or add other dict parameter if there can be more than 1 controller
https://github.com/jontofront/ecoNET-300-Home-Assistant-Integration/blob/master/custom_components/econet300/common.py
`controller_id = coordinator.controllerID`